### PR TITLE
Use RegexGenerator in applicable tests

### PR DESCRIPTION
**Pull request by Clockwork-Muse**
Use RegexGenerator where possible in tests.

Also added to `System.Private.DataContractSerialization`

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerTest.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public class ConsoleLoggerTest
             var defaultMonitor = new TestFormatterOptionsMonitor<SimpleConsoleFormatterOptions>(simpleOptions ?? new SimpleConsoleFormatterOptions());
             var systemdMonitor = new TestFormatterOptionsMonitor<ConsoleFormatterOptions>(systemdOptions ?? new ConsoleFormatterOptions());
             var jsonMonitor = new TestFormatterOptionsMonitor<JsonConsoleFormatterOptions>(jsonOptions ?? new JsonConsoleFormatterOptions());
-            var formatters = new List<ConsoleFormatter>() { 
+            var formatters = new List<ConsoleFormatter>() {
```
Comment (stephentoub) [inline src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerTest.cs:33]: Ditto

File: src/libraries/System.IO.Ports/tests/System.IO.Ports.Tests.csproj [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreAppCurrent)-windows;$(NetCoreAppCurrent)-Linux;$(NetCoreAppCurrent)-OSX;$(NetCoreAppCurrent)-FreeBSD;$(NetFrameworkMinimum)</TargetFrameworks>
+    <EnableRegexGenerator>true</EnableRegexGenerator>
```
Comment (stephentoub) [inline src/libraries/System.IO.Ports/tests/System.IO.Ports.Tests.csproj:4]: Note the TargetFrameworks line above. As it contains things other than NetCoreAppCurrent, the changes in this project won't work.

**Review by stephentoub — CHANGES_REQUESTED** [review]
Thanks for what's obviously a lot of work on this.

I starting going through it and left some comments, but the short of it is that I think most of the changes in here should be reverted, and we should only keep:
1. The src changes for DataContractSerializer
2. Changes where you made better use of built-in xunit helpers, like places you switched to use Assert.Matches(...) rather than Assert.True(Regex.IsMatch(...))

Many of the other changes to test projects won't work because those test projects also build for downlevel surface area, and the regex generator requires .NET 7 or newer surface area.  And for others that do work, I think many of the changes negatively impact the clarity of the tests, and I think that's more important than any startup benefits that might come from using the source generator in the tests.  The other benefit would be if we think these uses of the source generator are filling gaps from our dedicated regex test suite, but if that's the case, we should augment the test suite.

File: src/libraries/System.IO.Ports/tests/System.IO.Ports.Tests.csproj (hunk 2 above)
Comment (Clockwork-Muse) [reply src/libraries/System.IO.Ports/tests/System.IO.Ports.Tests.csproj:4]: I know - I'd made a pass previously, but seem to have been constrained for time, and forgot I hadn't finished it.

**Comment by Clockwork-Muse** [comment]
@stephentoub - Run seems to have failed due to OOM, but I'm not sure which/if I should just launch an azp run.

**Comment by danmoseley** [comment]
The out of memory is https://github.com/dotnet/runtime/issues/65791.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**